### ISSUE TITLE
Confirmation Before Pinning message

### ISSUE THREAD

**Issue by Akshun-01** [RocketChat/EmbeddedChat#450]
As an EmbeddedChat developer

**I need to**: 
Add a confirmation popup before pinning a message. 
![image](https://github.com/RocketChat/EmbeddedChat/assets/93793691/3c9eee1c-7710-451e-8be3-61995adfdf0d)

**So That**:
The user knows pinned messages are visible for everyone to see. It increases the user experience.
And as it is an extension of RocketChat, it would make it more similar and consistent.

**Acceptance Criteria**
- [ ] Check pin status for message and give popup accordingly
- [ ] on confirmation pin the message otherwise return

**Comment by Spiral-Memory** [comment]
Hey @Akshun-01 , would it be a good feature to have in embedded chat? I mean, I feel this app is mainly for integration into different platforms to streamline workflows. Adding an extra modal to confirm whether we want to pin may not be a very good idea. I think with embedded chat, we have a slightly different purpose, so just replicating features from Rocket Chat might not be suitable. However, it's just my opinion. The maintainers will know better. You can ask them if you want.

**Comment by Spiral-Memory** [comment]
Yes, @Akshun-01 , I understand. However, I still feel that for the main Rocket Chat app, it might be necessary in the situation you discussed. Still, I believe embedded chat is meant to make the workflow easier. Adding another modal for this wouldn't be very beneficial from my point of view.

@abhinavkrin , what do you think about this? Would it be good to add this feature to embedded chat as well?
